File: src/main/java/hilfling/backend/hilfling/controllers/AlbumController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.Album;
import hilfling.backend.hilfling.service.AlbumService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/albums")
public class AlbumController extends GenericBaseControllerImplementation<Album> {
    @Autowired
    private AlbumService service;
    public AlbumService getService() {
        return service;
    }

}

File: src/main/kotlin/hilfling/backend/hilfling/controller/AlbumController.kt
package hilfling.backend.hilfling.controller

import no.fg.hilflingbackend.model.Album
import no.fg.hilflingbackend.repository.AlbumRepository
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/albums")
class AlbumController {
    val repository = AlbumRepository()

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Int) : Album? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAll() : List<Album> {
        return repository.findAll()
    }

    @PostMapping
    fun create(
            @RequestParam("title") title: String,
            @RequestParam("isAnalog") isAnalog: Boolean
    ): Album {
        return repository.create(Album{

        })
    }
}